packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -60,14 +62,14 @@ function GridEditSingleSelectCell(props: GridEditSingleSelectCellProps) {
     isProcessingProps,
     error,
     onValueChange,
+    initialOpen = rootProps.editMode === 'cell',
```

Comment:
We have an enum for edit modes.

```suggestion
    initialOpen = rootProps.editMode === GridEditModes.Cell,
```

Reply:
Made suggested change